diff --git a/src/Controller/AdminControllerTrait.php b/src/Controller/AdminControllerTrait.php
index 0799386a5..172eff124 100644
--- a/src/Controller/AdminControllerTrait.php
+++ b/src/Controller/AdminControllerTrait.php
@@ -398,7 +398,7 @@ protected function searchAction()
     /**
      * The method that is executed when the user performs a 'batch' action to any entity.
      */
-    protected function batchAction(): RedirectResponse
+    protected function batchAction(): Response
     {
         $batchForm = $this->createBatchForm($this->entity['name']);
         $batchForm->handleRequest($this->request);
@@ -407,7 +407,10 @@ protected function batchAction(): RedirectResponse
             $actionName = $batchForm->get('name')->getData();
             $actionIds = $batchForm->get('ids')->getData();

-            $this->executeDynamicMethod($actionName.'<EntityName>BatchAction', [$actionIds, $batchForm]);
+            $methodResult = $this->executeDynamicMethod($actionName.'<EntityName>BatchAction', [$actionIds, $batchForm]);
+            if ($methodResult instanceof Response) {
+                return $methodResult;
+            }
         }

         return $this->redirectToReferrer();
